commands/src/main/java/org/wildfly/extras/creaper/commands/messaging/AbstractTransportConfigAddCommand.java

```diff
@@ -0,0 +1,249 @@
+package org.wildfly.extras.creaper.commands.messaging;
+
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implementation of common properties of connector and acceptor in messaging connections.
+ */
+abstract class AbstractTransportConfigAddCommand implements OnlineCommand {
+
+    protected final TransportConfigType configType;
+    protected final TransportConfigItem configItem;
+    protected final String name;
+    protected final String serverName;
+    protected final Map<String, String> params;
+    protected final boolean replaceExisting;
+
+    //in-vm
+    protected final Integer serverId;
+
+    //generic and remote
+    protected final String socketBinding;
+
+    //remote
+    protected final String factoryClass;
+
+
+    private AbstractTransportConfigAddCommand(TransportConfigType configType, TransportConfigItem configItem,
+                                             String name, String serverName, Map<String, String> params,
+                                             boolean replaceExisting, Integer serverId, String socketBinding,
+                                             String factoryClass) {
+        this.configType = configType;
+        this.configItem = configItem;
+        this.name = name;
+        this.serverName = serverName;
+        this.params = params;
+        this.replaceExisting = replaceExisting;
+        this.serverId = serverId;
+        this.socketBinding = socketBinding;
+        this.factoryClass = factoryClass;
+    }
+
+    protected AbstractTransportConfigAddCommand(InVmBuilder builder, TransportConfigItem configItem) {
```

Comment:
Can you parametrize reference to generic type with wildcard (i.e. `InVmBuilder<?> builder`) to prevent in this case IMO uneccessary compiler complaints?